hack/mock-unittest-data/gomod.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+import json
+import subprocess
+from contextlib import contextmanager
+from functools import partial
+from itertools import chain
+from os import environ
+from pathlib import Path
+from shutil import copyfile
+from shutil import rmtree as delete_directory_with_contents
+from sys import argv
+from tempfile import gettempdir
+
+from yarn import print_banner
+
+
+def mkdir_p(dirname):
+    return Path(dirname).mkdir(exist_ok=True, parents=True)
+
+
+def generate_auxiliary_dirs(mocked_data_dir):
+    mkdir_p(f"{mocked_data_dir}/non-vendored")
+    mkdir_p(f"{mocked_data_dir}/vendored")
+    mkdir_p(f"{mocked_data_dir}/workspaces")
+    mocked_data_dir_abspath = Path(f"{mocked_data_dir}").absolute()
```

Comment:
This function is pretty much the only place that works with `mocked_data_dir` as a relative path, the rest of the script uses the absolute path. You can just use the relative paths everywhere, you don't need the absolute path, it'll make the code simpler in terms of not reducing the number of arguments passed around, keep reading on I'll comment elsewhere....